wagtail/test/utils/page_tests.py

```diff
@@ -148,3 +162,288 @@ def assertAllowedParentPageTypes(self, child_model, parent_models, msg=None):
+            },
+        )
+        raise self.failureException(msg)
+
+    def assertPageIsEditable(
+        self,
+        page: Page,
+        post_data: Optional[Dict[str, Any]] = None,
+        user: Optional[AbstractBaseUser] = None,
+        msg: Optional[str] = None,
+    ):
+        """
+        Asserts that the page edit view works for ``page`` without raising a fatal error.
+
+        When ``user`` is provided, the test is conducted with them as the active user. Otherwise, a superuser is created and used for the test.
+
+        After a successful ``GET`` request, a ``POST`` request is made with field data in the request body. If ``post_data`` is provided, that will be used for this purpose. If not, this data will be extracted from the ``GET`` response HTML.
+        """
+        if user:
+            # rule out permission issues early on
+            if not page.permissions_for_user(user).can_edit():
+                self._formatMessage(
+                    msg,
+                    'Failed to load edit view for %(page_type)s "%(page)s":\nUser "%(user)s" have insufficient permissions.'
+                    % {
+                        "page_type": type(page).__name__,
+                        "page": page,
+                        "user": user,
+                    },
+                )
+                raise self.failureException(msg)
+        else:
+            if not hasattr(self, "_pageiseditable_superuser"):
+                self._pageiseditable_superuser = self.create_superuser(
+                    "assertpageiseditable"
+                )
+            user = self._pageiseditable_superuser
+
+        self.client.force_login(user, AUTH_BACKEND)
+
+        path = reverse("wagtailadmin_pages:edit", kwargs={"page_id": page.id})
+        try:
+            response = self.client.get(path)
+        except Exception as e:
+            self.client.logout()
+            msg = self._formatMessage(
+                msg,
+                'Failed to load edit view via GET for %(page_type)s "%(page)s":\n%(exc)s'
+                % {"page_type": type(page).__name__, "page": page, "exc": e},
+            )
+            raise self.failureException(msg)
+        if response.status_code != 200:
+            self.client.logout()
+            msg = self._formatMessage(
+                msg,
+                'Failed to load edit view via GET for %(page_type)s "%(page)s":\nReceived response with HTTP status code: %(code)s.'
+                % {
+                    "page_type": type(page).__name__,
+                    "page": page,
+                    "code": response.status_code,
+                },
+            )
+            raise self.failureException(msg)
+
+        if post_data is not None:
+            data_to_post = post_data
+        else:
+            data_to_post = querydict_from_html(
+                response.content.decode(), form_id="page-edit-form"
+            )
+            data_to_post["action-publish"] = ""
```

Comment:
question: should we care whether the provided user has permission to publish? Also, the docs example include "action-publish" in the post data.

Reply:
I see what you mean. If we are checking the permission up-front to reduce confusion later in the test, it makes sense that we check for publish permission. I'll submit a PR for this when I'm back from hols :)